src/H5Dmpio.c

```diff
@@ -636,8 +633,8 @@ H5D__mpio_opt_possible(H5D_io_info_t *io_info)
         }
 
         /* Check whether these are both simple or scalar dataspaces */
-        if (!((H5S_SIMPLE == H5S_GET_EXTENT_TYPE(mem_space) ||
-               H5S_SCALAR == H5S_GET_EXTENT_TYPE(mem_space)) &&
+        if (!((H5S_SIMPLE == H5S_GET_EXTENT_TYPE(mem_space) || H5S_SCALAR == H5S_GET_EXTENT_TYPE(mem_space) ||
+               H5S_NULL == H5S_GET_EXTENT_TYPE(mem_space)) &&
```

Comment:
From what I can tell, it should be fine to also perform optimized parallel I/O when the memory space is H5S_NULL. This would be the case when no elements are selected in the file space and a NULL dataspace or H5S_BLOCK was passed for the memory space. In either case, the intention would be to not contribute directly to the I/O, but still participate in collective operations.

Reply:
Is there a test for this case?

Reply:
Also shouldn't this case be added to the file space as well?

Reply:
Since `H5S_BLOCK` is currently not allowed for the file space, I only modified the memory space side of things. I imagine this should also be allowed for the case where a NULL file space is explicitly passed to `H5Dwrite`/`H5Dread`, but I don't believe we test this and the parallel compression tests are currently only testing the case of a NULL space getting created for the memory space when `H5S_BLOCK` is passed and no elements are selected in the dataspaces.